eng/common/scripts/Verify-RestApiSpecLocation.ps1

```diff
@@ -0,0 +1,281 @@
+  }
+  catch {
+    LogError "ServiceDir:$ServiceDirectory, PackageName:$PackageName. Failed to search commit $commit with exception:`n$_ "
+    exit 1
+  }
+}
+
+function Verify-YamlContent([string]$markdownContent) {
+  $splitString = '``` yaml|```yaml|```'
+  $yamlContent = $markdownContent -split $splitString
+  foreach($yamlSection in $yamlContent) {
+    if ($yamlSection) {
+      try {
+        # remove the lines like: $(tag) == 'package-preview-2023-09'
+        $yamlSection = $yamlSection -replace '^\s*\$\(.+\)\s*==.+', ''
+        $yamlobj = ConvertFrom-Yaml -Yaml $yamlSection
+        if($yamlobj) {
+          $batchValue = $yamlobj["batch"]
+          $requireValue = $yamlobj["require"]
+          $inputFileValue = $yamlobj["input-file"]
+          if ($requireValue) {
+            LogDebug "ServiceDir:$ServiceDirectory, PackageName:$PackageName. 'require' is set as:$requireValue"
+            Verify-Url $requireValue
+          }
+          elseif ($inputFileValue) {
+            LogDebug "ServiceDir:$ServiceDirectory, PackageName:$PackageName. 'input-file' is set as:$inputFileValue"
+            foreach($inputFile in $inputFileValue) {
+              Verify-Url $inputFile
+            }
+          }
+          elseif ($batchValue) {
+            # there are some services which use batch mode for sdk generation, e.g. Azure.AI.Language.QuestionAnswering
+            foreach($batch in $batchValue) {
+              $requireValue = $batch["require"]
+              $inputFileValue = $batch["input-file"]
+              if ($requireValue) {
+                LogDebug "ServiceDir:$ServiceDirectory, PackageName:$PackageName. 'require' is set as:$requireValue"
+                Verify-Url $requireValue
+              }
+              elseif ($inputFileValue) {
+                LogDebug "ServiceDir:$ServiceDirectory, PackageName:$PackageName. 'input-file' is set as:$inputFileValue"
+                foreach($inputFile in $inputFileValue) {
+                  Verify-Url $inputFile
+                }
+              }
+            }
+          }
+        }
+      }
+      catch {
+        Write-Host "ServiceDir:$ServiceDirectory, PackageName:$PackageName. Failed to parse yaml section $yamlSection with exception:`n$_ "
+      }
+    }
+  }
+}
+
+function Verify-PackageVersion() {
+  try {
+    $packages = @{}
+    if ($FindArtifactForApiReviewFn -and (Test-Path "Function:$FindArtifactForApiReviewFn"))
+    {
+      $packages = &$FindArtifactForApiReviewFn $ArtifactLocation $PackageName
+    }
+    else
+    {
+      LogError "The function for 'FindArtifactForApiReviewFn' was not found.`
+      Make sure it is present in eng/scripts/Language-Settings.ps1 and referenced in eng/common/scripts/common.ps1.`
+      See https://github.com/Azure/azure-sdk-tools/blob/main/doc/common/common_engsys.md#code-structure"
+      exit 1
+    }
+    if (-not $PackageInfoDirectory)
+    {
+      $PackageInfoDirectory = Join-Path -Path $ArtifactLocation "PackageInfo"
```

Comment:
Is this only for local testing?

Reply:
No. This is used in release pipeline where doesn't pass in the value. Because during release job stage, it doesn't have the `packageInfo.json` generated at signed artifact folder. This script needs to generate that file before getting the package version.